samples/create_event.cc

```diff
@@ -1,8 +1,111 @@
 #include "proto/spec.pb.h"
+#include "third_party/statusor/statusor.h"
+#include <fstream>
 #include <iostream>
-int main() {
+
+void GetUserInput(std::string prompt, std::string& res) {
+    std::cout << prompt << ": ";
+    getline(std::cin, res);
+}
+
+void OutputToInterface(std::string content, bool isError = false) {
+    if (isError) {
+        std::cerr << content << std::endl;
+    } else {
+        std::cout << content << std::endl;
+    }
+}
+
+absl::Status ValidateFile(std::string file_path){
+    std::fstream input(file_path, std::ios::in | std::ios::binary);
+    if (input) {
+        std::string cont;
+        GetUserInput("File already exists. Overwrite? (y/n)", cont);
+        if (cont != "y"){
+            return absl::CancelledError();
+        }
+    }
+    return absl::OkStatus();
+}
+
+
+absl::Status WriteToFile(std::string file_path, io::cloudevents::v1::CloudEvent* event){
+    std::fstream output(file_path, std::ios::out | std::ios::trunc | std::ios::binary);
+    if (!event -> SerializeToOstream(&output)) {
+        return absl::InvalidArgumentError("Could not write to file given.");
+    }
+    return absl::OkStatus();
+}
+
+absl::Status PopulateEvent(io::cloudevents::v1::CloudEvent* event){
+    // TODO (#7): Abstract this out to a Buider with validation.
+
+    GetUserInput("Enter id", *event -> mutable_id());
+    GetUserInput("Enter source", *event -> mutable_source());
+    GetUserInput("Enter spec_version", *event -> mutable_spec_version());
+    GetUserInput("Enter type", *event -> mutable_type());
+
+    std::string has_data;
+    GetUserInput("Would you like to enter data (y/n)", has_data);
+    if (has_data=="y") {
+        std::string data_type;
+        // TODO (#6): Support Any data
+        GetUserInput("Enter data type (bytes/ string)", data_type);
+        const static std::unordered_map<std::string,int> data_type_to_case{
+            {"bytes",1},
+            {"string",2},
+            {"other", 3}
+        };
+
+        if (data_type_to_case.find(data_type) == data_type_to_case.end()) {
+            return absl::InvalidArgumentError("Data type not recognized.");
+        }
+        
+        switch(data_type_to_case.at(data_type)){
+            case 1: GetUserInput("Enter data", *event -> mutable_binary_data()); break;
+            case 2: GetUserInput("Enter data", *event -> mutable_text_data()); break;
+            case 3: return absl::UnimplementedError("Other data type not supported yet.");
+            default: {
+                return absl::InternalError("Data type mapping resulted in unknown case.");
+            }
+        }
+    }
+    return absl::OkStatus();
+}
+
+int main(int argc, char* argv[]) {
     GOOGLE_PROTOBUF_VERIFY_VERSION;
+    absl::Status program_status;
+    
+    // ensure that a write file is specified
+    if (argc != 2) {
```

Comment:
I'm not sure why we are using command line arguments instead of flags (https://abseil.io/docs/cpp/guides/flags)?

If this is intended, the condition should be `argc>2`